src/gsi/crtm_interface.f90

```diff
@@ -1822,15 +1874,34 @@ subroutine call_crtm(obstype,obstime,data_s,nchanl,nreal,ich, &
      end do
   endif
 
+
+  ! Calculate Thompson effective radius for each hydrometeor 
+  if ( icmask .and. n_clouds_fwd_wk > 0 .and. imp_physics==08 .and. lprecip_wk) then
+     do ii = 1, n_clouds_fwd_wk
+        iii=jcloud(ii)
+        call calc_thompson_reff(rho_air,h,cloud(:,ii),cloud_names(iii),cloudefr(:,ii))
+     end do
+  endif
+
   ! Calculate GFDL cloud fraction (if no cf in metguess table) based on PDF scheme 
-! if ( icmask .and. n_clouds_fwd_wk > 0 .and. imp_physics==11 .and.  lcalc_gfdl_cfrac ) then
+  ! if ( icmask .and. n_clouds_fwd_wk > 0 .and. imp_physics==11 .and.  lcalc_gfdl_cfrac ) then
   if ( icmask .and. n_clouds_fwd_wk > 0 .and. imp_physics==11 .and.  lprecip_wk ) then
      cf_calc  = zero
      call calc_gfdl_cloudfrac(rho_air,h,q,cloud,hs,garea,qs,cf_calc)
      cf   = cf_calc
      icfs = 0        ! load cloud fraction into CRTM 
   endif
 
+  ! Calculate Thompson cloud fraction 
+  if ( icmask .and. n_clouds_fwd_wk > 0 .and. imp_physics==08 .and.  lprecip_wk ) then
+     lmfdeep2 = .true.     
+     cf_calc  = zero
+         ! sum of the cloud condensate amount for all 5 hydrometeos (ql + qi + qs + qg + qh + qr) 
+         qcond(:) = cloud(:,1) + cloud(:,2) + cloud(:,3) + cloud(:,4) + cloud(:,5)
```

Comment:
Are these lines in different levels?

Reply:
Code should be aligned inside if-then blocks.